File: js/Pedido.js
class Pedido{
    constructor(id, end, cliente){
        this.id = id;
        this.end = end;
        this.products = [];
        this.cliente = cliente
    }

    addProduto(prod){
        this.products.push(prod);
    }

    sumPrice(){
        let sum = 0;
        this.products.forEach(item => sum += item.price);
        return sum;
    }

    imprimir(){
        console.log(`
        id do pedido: ${this.id}
        Endereço do pedido: ${this.end}
        Nome do cliente: ${this.cliente.name}

        =====    Produtos    =====
        `)
        this.products.forEach(item => {
            console.log(`
        ${item.name} - ${item.cat.name} - R$ ${item.price.toFixed(2)}
            `)
        });
    }
}

export { Pedido }

File: py/03-Usanda-Uma-Classe-Em-Outra/Pedido.py
class Pedido:

    def __init__(self, id, end, cliente):
        self.id = id
        self.end = end
        self.cliente = cliente
        self.produtos = []
        
    def addProduto(self, item):
        self.produtos.append(item)
    
    def sumPrice(self):
        if len(self.produtos) == 0:
            print('Pedido vazio.')
        else:
            soma = 0
            for produto in self.produtos:
                soma += produto.preco
            return soma
      
    def imprimir(self):
        print(f"id pedido: {self.id}")
        print(f"cliente: {self.cliente}")
        print(f"endereço: {self.end}")
        print("\n\n==== itens ====")
        for produto in self.produtos:
            print(produto.nome)
            print(f"{produto.preco} - {produto.cat.nome}")
